### PR TITLE
Clamp Lpt to ngraph

### PR DESCRIPTION
**Pull request by v-Golubev**

### CODE REVIEW

File: inference-engine/tests/functional/inference_engine/lp_transformations/clamp_transformation.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+const std::vector<ngraph::Shape> shapes = {
+    { 1, 32, 72, 48 },
+    { 1, 3, 224, 224 }
+};
+
+const std::vector<ClampTransformationTestValues> testValues = {
+    { LayerTransformation::createParamsU8I8(), { 128 }, { 3.0f } },
+    { LayerTransformation::createParamsI8I8(), { 2 }, { -5.0f } }
+};
```
Comment (eshoguli) [inline inference-engine/tests/functional/inference_engine/lp_transformations/clamp_transformation.cpp:94]: can you add additional test cases:
1. per-channel quantization
2. updatePrecision = false (no INT8 tensors in model)

File: inference-engine/tests/ngraph_functions/src/low_precision_transformations/clamp_function.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        ngraph::ResultVector results{ std::make_shared<ngraph::opset1::Result>(clamp) };
+        return std::make_shared<ngraph::Function>(results, ngraph::ParameterVector{ input }, "ClampTransformation");
+    }
+
+    std::shared_ptr<ngraph::Function> ClampFunction::getReference(
+        const ngraph::element::Type originalFunctionPrecision,
+        const ngraph::Shape& inputShape,
+        const ExpectedValues& values) {
+        auto input = std::make_shared<ngraph::opset1::Parameter>(originalFunctionPrecision, ngraph::Shape(inputShape));
```
Comment (eshoguli) [inline inference-engine/tests/ngraph_functions/src/low_precision_transformations/clamp_function.cpp:49]: use `values.lowPrecision` instead `originalFunctionPrecision`. Cases:
1. low precision is expected if LayerTransformation::Param::updatePrecisions = true
2. original function precision is expected if LayerTransformation::Param::updatePrecisions = false

Note, please, don't pass LayerTransformation::Param instance into test

File: inference-engine/src/transformations/src/transformations/low_precision/clamp.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    Shape multiplyShape = dequantization.multiply->get_input_shape(1);
+    if (!multiplyShape.empty() && std::any_of(multiplyShape.cbegin(), multiplyShape.cend(), [](const size_t value) { return value != 1U; })) {
+        return false;
+    }
+
+    return true;
+}
+
+bool ClampTransformation::isPrecisionPreserved(std::shared_ptr<Node> layer) const noexcept {
+    return true;
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/transformations/src/transformations/low_precision/clamp.cpp]: Done, thanks!

File: inference-engine/src/transformations/src/transformations/low_precision/clamp.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2018-2020 Intel Corporation
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/transformations/src/transformations/low_precision/clamp.cpp]: Done, thanks!

File: inference-engine/tests/functional/inference_engine/lp_transformations/clamp_transformation.cpp [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                testValues.subtractValues,
+                testValues.multiplyValues
+            });
+    }
+
+    static std::string getTestCaseName(testing::TestParamInfo<ClampTransformationParams> obj) {
+        const ngraph::element::Type precision = std::get<0>(obj.param);
+        const ngraph::Shape shape = std::get<1>(obj.param);
+        const ClampTransformationTestValues testValues = std::get<2>(obj.param);
+        return LayerTransformation::getTestCaseNameByParams(precision, shape, testValues.params);
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/tests/functional/inference_engine/lp_transformations/clamp_transformation.cpp]: Done, thanks!

File: inference-engine/src/transformations/include/transformations/low_precision/clamp.hpp [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+#pragma once
+
+#include "transparent_base_transformation.hpp"
+
+namespace ngraph {
+namespace pass {
+namespace low_precision {
+
+class TRANSFORMATIONS_API ClampTransformation : public TransparentBaseTransformation {
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/transformations/include/transformations/low_precision/clamp.hpp]: Done, thanks!

File: inference-engine/tests/ngraph_functions/include/ngraph_functions/low_precision_transformations/clamp_function.hpp [hunk 7]
```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <ngraph/ngraph.hpp>
+#include "common/fake_quantize_on_data.hpp"
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/tests/ngraph_functions/include/ngraph_functions/low_precision_transformations/clamp_function.hpp]: Done, thanks!

File: inference-engine/tests/ngraph_functions/include/ngraph_functions/low_precision_transformations/clamp_function.hpp [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+#include "transformations/low_precision/layer_transformation.hpp"
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/tests/ngraph_functions/include/ngraph_functions/low_precision_transformations/clamp_function.hpp:7]: Done, thanks!

File: inference-engine/tests/functional/inference_engine/lp_transformations/clamp_transformation.cpp (hunk 1 above)
Comment (v-Golubev) [reply inference-engine/tests/functional/inference_engine/lp_transformations/clamp_transformation.cpp:158]: Done, thanks!

File: inference-engine/tests/ngraph_functions/src/low_precision_transformations/clamp_function.cpp (hunk 2 above)
Comment (v-Golubev) [reply inference-engine/tests/ngraph_functions/src/low_precision_transformations/clamp_function.cpp]: Done, thanks!

File: inference-engine/src/transformations/src/transformations/low_precision/clamp.cpp [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        max += shift[0];
+    }
+
+    auto replacement = std::make_shared<ngraph::opset1::Clamp>(newClamp->get_input_node_shared_ptr(0), min, max);
+    replace_node(newClamp, replacement);
+
+    element::Type outputClampType = dequantization.multiply ?
+        dequantization.multiply->get_output_element_type(0) :
+        dequantization.subtract->get_output_element_type(0);
+    ngraph::pass::low_precision::NetworkHelper::setOutDataPrecision(replacement, outputClampType);
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply inference-engine/src/transformations/src/transformations/low_precision/clamp.cpp:53]: Discussed offline, let keep it as is

**Comment by v-Golubev** [comment]
@eshoguli Comments fixed. CPU and GPU plugin tests were added. Review it, please.

**Pull request closed without merging**